tests/functional/trace_io_events.py

```diff
@@ -0,0 +1,157 @@
+                                   f" -l {int(discard_length.get_value())} {disk.system_path}")
+        with TestRun.step("Stop tracing"):
+            iotrace.stop_tracing()
+        with TestRun.step("Verify trace correctness"):
+            trace_path = iotrace.get_latest_trace_path()
+            events = iotrace.get_trace_events(trace_path)
+            events_parsed = iotrace.parse_json(events)
+            result = any(
+                'io' in event and
+                'operation' in event['io'] and
+                event['io']['operation'] == 'Write' and
+                # LBA 0 events don't have a lba field, so skip them
+                'lba' in event['io'] and
+                int(event['io']['lba']) == int(2 * write_length.get_value() / 512) and
+                int(event['io']['len']) == int(write_length.get_value() / 512) and
+                f"/dev/{event['device']['name']}" == disk.system_path for
+                event in events_parsed)
+            if not result:
+                raise Exception("Could not find write event")
+            result = any(
+                'io' in event and
+                # Read events don't have an operation field
+                'operation' not in event['io'] and
+                'lba' in event['io'] and
+                int(event['io']['lba']) == int(2 * read_length.get_value() / 512) and
+                int(event['io']['len']) == int(read_length.get_value() / 512) and
+                f"/dev/{event['device']['name']}" == disk.system_path for
+                event in events_parsed)
+            if not result:
+                raise Exception("Could not find read event")
+            result = any(
+                'io' in event and
+                'operation' in event['io'] and
+                event['io']['operation'] == 'Discard' and
+                'lba' in event['io'] and
+                int(event['io']['lba']) == int(2 * discard_length.get_value() / 512) and
+                int(event['io']['len']) == int(discard_length.get_value() / 512) and
+                f"/dev/{event['device']['name']}" == disk.system_path for
+                event in events_parsed)
+            if not result:
+                raise Exception("Could not find discard event")
+
+
+def test_lba_histogram():
+    TestRun.LOGGER.info("Testing lba histogram")
+    iotrace = TestRun.plugins['iotrace']
+    number_buckets = 32
+    bucket_size = 256
+    start_lba = 10240
+    end_lba = start_lba + number_buckets * bucket_size
+    for disk in TestRun.dut.disks:
+        io_len = Size(1, disk.block_size)
+        with TestRun.step("Start tracing"):
+            iotrace.start_tracing([disk.system_path])
+            time.sleep(5)
+        with TestRun.step("Send write commands"):
+            for bucket_index in range(number_buckets):
+                for i in range(bucket_index + 1):
+                    seek = (start_lba + bucket_index * bucket_size + randrange(bucket_size)) * 512 / disk.block_size.get_value()
```

Comment:
Please extract 512 to e.g. variable or helper function with comment on why is it needed (iotrace to different block size conversion)